### DISCUSSION TITLE
When running my app, I get the error "Unable to Check For Updates"

### DISCUSSION THREAD

**Discussion by jmshgs** [sparkle-project/Sparkle#2499] [Q&A] [answered]
## Summary

When running my app, this error pops up. 
<img src="https://github.com/sparkle-project/Sparkle/assets/72482214/dbaef42c-214a-473a-8930-da339101abe0" width="400">

In the console it reads:
<img width="850" alt="image" src="https://github.com/sparkle-project/Sparkle/assets/72482214/8af16530-7a5a-4b5f-842d-52d3ca110a5c">

This is strange because the app is using Sparkle 2 and doesn't have an EdDSA key associated with it.

<img width="850" alt="image" src="https://github.com/sparkle-project/Sparkle/assets/72482214/5e514c77-d801-4e9a-a600-e36ba6bfd52e">

## Version

this is using sparkle 2.5.2

**Comment by zorgiepoo** [comment, marked as answer]
Make sure you copy/pasted the SUPublicEDKey correctly. That error is logged when the input is not proper base64 encoding (which I think is the case here) or when the decoded length after base64 conversion is wrong.

Run generate_keys to get the previously generated public key again.

**Reply by jmshgs** [reply]
looks like we forgot the `=` at the end. i should've noticed. thanks!

**Comment by jpmhouston** [comment]
Forgive resurrecting this thread, but I have a question about the same alert. What are the other conditions that cause it to appear? By any chance will at open if one asks for a channel that doesn't yet exist in any released versions in the appcast?

I was about to investigate why I'm seeing this alert, and one thing I did was add support for a beta channel. Although I also brought all my source into a new repo and maybe the SUFeedURL is wrong. In my case I don't think it's SUPublicEDKey which I believe I didn't change.

Once I discover why I'm seeing this same alert, I'll post the answer here. Anyone chiming in before then is appreciated though.

**Reply by zorgiepoo** [reply]
Check the console logs like the originator did here.

The code is at https://github.com/sparkle-project/Sparkle/blob/2.x/Sparkle/SPUUpdater.m#L217C9-L217C52 (`checkIfConfiguredProperlyAndRequireFeedURL:validateXPCServices:error:`)

The feed URL is not required to be present when starting the updater although it's recommended to be present. If the feed URL is there, there is validation on it. There's no validation for channels in there that I see.

**Reply by jpmhouston** [reply]
Yes, all the errors leading to this alert indeed originate in the `checkIfConfiguredProperly...` method in SPUUpdater.m  you mentioned and have log messages for each failure condition.

To anyone else getting this alert, yes, find the message logged and if you're need to dive into the Sparkle source code for more context, then start at this method.

I finally dug back through the saved console logs in Xcode's Report Navigator sidebar tab and for me it was "Sparkle cannot target a bundle that does not have a valid version". This explains why after fixing my build phase script that patches the build number, I was no longer getting the alert. My script was working in my old repo, but some detail about the new one was making the script maybe set CFBundleVersion to 0 or removing it altogether IDK.
